apps/lib/apps_ui.c

```diff
@@ -72,7 +72,7 @@ static int ui_write(UI *ui, UI_STRING *uis)
             {
                 const char *password =
                     ((PW_CB_DATA *)UI_get0_user_data(ui))->password;
-                if (password && password[0] != '\0')
```

Comment:
I am assuming someone put empty string check in for a reason..
Should we be taking it out?

Reply:
> I am assuming someone put empty string check in for a reason..
> Should we be taking it out?

As far as I recall, I removed this check for consistency with commit ca3245a61989009a99931748723d12e30d0a66b2 by @mattcaswell.
I'm not 100% sure about it, but at least none of the tests break.

Reply:
If someone explicitly sets an empty password then the old code ignores it and tries to get a password from the fallback method. The fallback is usually stdin, unless stdin has been disabled using (for example) the "-batch" option - in which case it will just fail. But if someone has explicitly set an empty password then presumably we should use it. So this change looks ok to me.